cli/pcluster/config/validators.py

```diff
@@ -922,3 +927,78 @@ def base_os_validator(param_key, param_value, pcluster_config):
         )
 
     return [], warnings
+
+
+def queue_settings_validator(param_key, param_value, pcluster_config):
+    errors = []
+    cluster_section = pcluster_config.get_section("cluster")
+    scheduler = cluster_section.get_param_value("scheduler")
+
+    if scheduler != "slurm":
+        errors.append("queue_settings is supported only with slurm scheduler")
+
+    return errors, []
+
+
+def queue_validator(section_key, section_label, pcluster_config):
+    errors = []
+    section = pcluster_config.get_section(section_key, section_label)
+    compute_resource_labels = str(section.get_param_value("compute_resource_settings") or "").split(",")
+
+    instance_types = []
+    for compute_resource_label in compute_resource_labels:
+        compute_resource = pcluster_config.get_section("compute_resource", compute_resource_label)
+        if compute_resource:
+            instance_type = compute_resource.get_param_value("instance_type")
+            if instance_type in instance_types:
+                errors.append(
+                    "Duplicate instance type '{0}' found in queue '{1}'. "
+                    "Compute resources in the same queue must use different instance types".format(
+                        instance_type, section_label
+                    )
+                )
+            else:
+                instance_types.append(instance_type)
+    return errors, []
+
+
+def settings_validator(param_key, param_value, pcluster_config):
```

Comment:
optional: totally fine with leaving this as is but in theory it could have been implemented with a single regex
(at a very high level something similar to `[allowedchars]{1,64}(,[allowedchars{1,64}]){0,4}`) so that we were setting this in the mappings with the allowed_values feature.
Error would return the allowed regex by default if I'm not wrong so that updating the regex does not require changing all error messages.

Reply:
I was opting for this solution first but then I got the impression that the resulting error message would have been too cryptic.

Reply:
Agree however it is something common also in the AWS cli

Reply:
I see your point. If it was a single value I would have definitely opted for the regex. But since it's a multiple value, and the labels are used both in settings parameters and in sections, I thought it was better to direct more explicitely the user toward which label is wrong and must be changed (in both places).